freedomofpress/securedrop
# Rename journalist key routes

## Status

Ready for review

## Description of Changes

Fixes #4421 

Changes proposed in this pull request:

Rename `why-journalist-key` and `journalist-key` routes of the Source Interface to `why-public-key` and `public-key` respectively. This aligns the route names with the language used in the associated pages where "public key" is used to designate the SecureDrop Submission Key of the instance.

Also rename the functions associated with the routes.

Maintain the `journalist-key` and redirect it to `public-key` in case it is used or linked by instance owners or sources. Add a unit test to verify the redirect.

## Testing

How should the reviewer test this PR?
Write out any special testing steps here.

1. In the source interface, navigate to `/lookup`.
1. Hover over the *public key* link.
    - [ ] Verify the link points to `/public-key`.
1. Click the *public key* link.
    - [ ] Verify the key file downloads as expected.
1. Click the *learn more* link.
    - [ ] Verify the `/why-public-key` page loads.
1. Hover over the *Download* link.
    - [ ] Verify the link points to `/public-key`.
1. Click the *Download* link.
    - [ ] Verify the key file downloads as expected.
1. Using the browser address bar navigate to `/journalist-key`.
    - [ ] Verify the redirect works and the key file downloads as expected.

## Deployment

Any special considerations for deployment? Consider both:

1. Upgrading existing production instances.
2. New installs.

## Checklist

### If you made changes to the server application code:


### If you made changes to `securedrop-admin`:


### If you made changes to the system configuration:


### If you made non-trivial code changes:


Choose one of the following:

Ref.: https://github.com/freedomofpress/securedrop-docs/issues/111, https://github.com/freedomofpress/securedrop-docs/pull/112

### If you added or updated a code dependency:

Choose one of the following:
